src/utils.ts

```diff
@@ -1,3 +1,12 @@
+export const assertNotNull = <T>(obj: T | null, message: string) => {
```

Comment:
ts-utils is a dependency, and should have pretty good tree-shaking if you import from specific modules, is there a reason not to use it?

Reply:
It's only a dev dependency but if y'all are ok with it I can add it.
It also doesn't have assertNotNull but I could use assertInstanceOf instead.

Reply:
Added as a peerDependency